robocop/checkers/naming.py

```diff
@@ -662,3 +669,59 @@ def find_similar_variables(self, tokens, node):
                     col=token.col_offset,
                 )
             self.variables[normalized_token].add(token.value)
+
+
+class DeprecatedStatementChecker(VisitorChecker):
+    """Checker for finding deprecated statments."""
+
+    reports = (
+        "deprecated-statement",
+    )
+    # deprecated:alternative
+    deprecated_statements_rf3 = {}
+    deprecated_statements_rf4 = {"run keyword unless": "IF/ELSE",
+                                 "run keyword if":     "IF/ELSE"
+                                 }
+    deprecated_statements_rf5 = {"run keyword unless": "IF/ELSE"}
+
+    def visit_SuiteSetup(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.name, node)
+
+    def visit_TestSetup(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.name, node)
+
+    def visit_Setup(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.name, node)
+
+    def visit_SuiteTeardown(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.name, node)
+
+    def visit_TestTeardown(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.name, node)
+
+    def visit_Teardown(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.name, node)
+
+    def visit_TestCase(self, node):  # noqa
+        self.generic_visit(node)
+
+    def visit_KeywordCall(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.keyword, node)
+
+    def check_if_keyword_is_deprecated(self, keyword_name, node):
+        deprecated_statements = self.deprecated_statements_rf3
+        if ROBOT_VERSION.major == 4:
+            deprecated_statements = self.deprecated_statements_rf4
+        elif ROBOT_VERSION.major == 5:
+            deprecated_statements = self.deprecated_statements_rf5
+        if deprecated_statements.get(keyword_name.lower(), None):
```

Comment:
Keywords in RF are not only case insensitive but also can ignore spaces, underscores and can contain prefix with library name. Following are equal:
```
Run Keyword Unless
runkeywordunless
run_keyword_unless
BuiltIn.Run Keyword Unless
builtin.run_KeywoRD_UNLESS
```
But fortunately robocop have helper methods for that, you can use ``normalize_robot_name`` with ``remove_prefix=builtin.`` same as [here](https://github.com/MarketSquare/robotframework-robocop/blob/6c98345f3cb87ca916a00c191667a55a82e2d959/robocop/checkers/naming.py#L378)

Reply:
thank you, done